server/src/main/kotlin/org/jetbrains/teamcity/vault/server/VaultConnector.kt

```diff
@@ -72,31 +69,36 @@ class VaultConnector(dispatcher: EventDispatcher<BuildServerListener>, private v
         @JvmStatic
         fun revoke(info: LeasedWrappedTokenInfo, trustStoreProvider: SSLTrustStoreProvider, catch: Boolean = true): Boolean {
             val settings = info.connection
-            if (settings.auth.method == AuthMethod.AWS_IAM) {
-                // AWS IAM auth uses separate tokens, so we cannot revoke agent token from server
-                // we don't even request wrapped token before build
-                return true
-            }
-            assert(settings.auth.method == AuthMethod.APPROLE)
-            try {
-                val template = createRestTemplate(settings, trustStoreProvider)
-                // Login and retrieve server token
-                val (token, _) = performLogin(template, settings, extractTokenAndAccessor)
-
-                template.withVaultToken(token)
-                try {
-                    // Revoke agent token
-                    revokeAccessor(template, info.accessor, info.connection)
-                } finally {
-                    // Revoke server token we just obtained
-                    revokeSelf(template)
+            when (settings.auth.method) {
+                AuthMethod.APPROLE,
+                AuthMethod.LDAP,
+                AuthMethod.USERPASS -> {
+                    try {
+                        val template = createRestTemplate(settings, trustStoreProvider)
+                        // Login and retrieve server token
+                        val (token, _) = performLogin(template, settings, extractTokenAndAccessor)
+
+                        template.withVaultToken(token)
+                        try {
+                            // Revoke agent token
+                            revokeAccessor(template, info.accessor, info.connection)
+                        } finally {
+                            // Revoke server token we just obtained
+                            revokeSelf(template)
+                        }
+                        return true
+                    } catch (e: Exception) {
+                        LOG.warnAndDebugDetails("Failed to revoke token", e)
+                        if (!catch) throw e
+                    }
+                    return false
+                }
+                AuthMethod.AWS_IAM -> {
```

Comment:
Keeping this case first could simplify diff

Reply:
done